src/Features/CSharp/Portable/BraceCompletion/AbstractCurlyBraceOrBracketCompletionService.cs

```diff
@@ -173,9 +178,13 @@ static LinePosition GetIndentedLinePosition(ParsedDocument document, SourceText
 
             static ImmutableArray<TextChange> GetMergedChanges(TextChange newLineEdit, ImmutableArray<TextChange> formattingChanges, SourceText formattedText)
             {
+                var formattingChangeRanges = formattingChanges.SelectAsArray(f => f.ToTextChangeRange());
+                if (formattingChangeRanges.IsEmpty)
+                    return ImmutableArray.Create(newLineEdit);
```

Comment:
@jasonmalinowski tbh, I'm not sure how could we make the assumption that calling `FormatTrackingSpan` with a newline inserted would always return a non-empty change. Wouldn't it depend on user settings?

Reply:
@genlu I'm not sure I follow the question.

Reply:
We add a newline between brackets in some cases, and then pass the changed document to format the span between brackets. The assumption is the format change will never be empty as long as we inserted a newline, which I don't think it's guaranteed.

Reply:
yeah, i don't expect we would have that guarantee.  tbh, i really don't get what this code (including prior to your change) is trying to do and how it is going about trying to do it.  Do you feel confident we have good test coverage here to catch issues?

Reply:
OK, after a few attempts on tweaking the options, I finally get a test to repro the crash in both 17.5 branch (which is before https://github.com/dotnet/roslyn/pull/66575) and main

```cs
[WpfFact]
public void Test()
{
    var code = @"namespace NS1
$$";
    var expected = @"namespace NS1
{}";
    var expectedAfterReturn = @"namespace NS1
{

}";
    var globalOptions = new OptionsCollection(LanguageNames.CSharp)
    {
        { FormattingOptions2.SmartIndent, FormattingOptions2.IndentStyle.None },
        { AutoFormattingOptionsStorage.FormatOnCloseBrace, false },
    };
    using var session = CreateSession(code, globalOptions);
    Assert.NotNull(session);
    CheckStart(session.Session);
    Assert.Equal(expected, session.Session.SubjectBuffer.CurrentSnapshot.GetText());
    CheckReturn(session.Session, 0, expectedAfterReturn);
}
```


```
[xUnit.net 00:00:05.25]     Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.AutomaticCompletion.AutomaticBraceCompletionTests.Test [FAIL]
[xUnit.net 00:00:05.25]       System.ArgumentException : newChanges
[xUnit.net 00:00:05.25]       Stack Trace:
[xUnit.net 00:00:05.25]         src\Compilers\Core\Portable\InternalUtilities\TextChangeRangeExtensions.cs(129,0): at Roslyn.Utilities.TextChangeRangeExtensions.Merge(ImmutableArray`1 oldChanges, ImmutableArray`1 newChanges)
[xUnit.net 00:00:05.25]         src\Features\CSharp\Portable\BraceCompletion\AbstractCurlyBraceOrBracketCompletionService.cs(176,0): at Microsoft.CodeAnalysis.CSharp.BraceCompletion.AbstractCurlyBraceOrBracketCompletionService.<GetTextChangeAfterReturn>g__GetMergedChanges|5_2(TextChange newLineEdit, ImmutableArray`1 formattingChanges, SourceText formattedText)
[xUnit.net 00:00:05.25]         src\Features\CSharp\Portable\BraceCompletion\AbstractCurlyBraceOrBracketCompletionService.cs(153,0): at Microsoft.CodeAnalysis.CSharp.BraceCompletion.AbstractCurlyBraceOrBracketCompletionService.GetTextChangeAfterReturn(BraceCompletionContext context, IndentationOptions options, CancellationToken cancellationToken)
[xUnit.net 00:00:05.25]         src\EditorFeatures\Core\AutomaticCompletion\BraceCompletionSessionProvider.BraceCompletionSession.cs(290,0): at Microsoft.CodeAnalysis.AutomaticCompletion.BraceCompletionSessionProvider.BraceCompletionSession.PostReturn()
[xUnit.net 00:00:05.25]         src\EditorFeatures\TestUtilities\AutomaticCompletion\AbstractAutomaticBraceCompletionTests.cs(86,0): at Microsoft.CodeAnalysis.Editor.UnitTests.AutomaticCompletion.AbstractAutomaticBraceCompletionTests.CheckReturn(IBraceCompletionSession session, Int32 indentation, String result)
[xUnit.net 00:00:05.25]         src\EditorFeatures\CSharpTest\AutomaticCompletion\AutomaticBraceCompletionTests.cs(1694,0): at Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.AutomaticCompletion.AutomaticBraceCompletionTests.Test()
```